dynamic_programming/max_sub_array.py

```diff
@@ -64,6 +71,9 @@ def max_sub_array(nums: List[int]) -> int:
 
 
 if __name__ == "__main__":
+    """
+    A random simulation of this algorithm.
+    """
```

Comment:
Let's move the `matplotlib` import down here because it is only needed in the `__main__` section and that third party library won't be imported when this file is used as a library of functions.